attestation-service/attestation-service/src/policy_engine/mod.rs

```diff
@@ -35,38 +22,25 @@ impl PolicyEngineType {
     }
 }
 
-#[derive(Debug, Serialize, Deserialize, PartialEq)]
-pub struct PolicyListEntry {
-    pub id: String,
-    pub digest: PolicyDigestEntry,
-}
-
-#[derive(Debug, Serialize, Deserialize, PartialEq)]
-pub struct PolicyDigestEntry {
-    pub algorithm: String,
-    pub value: String,
-}
-
 type PolicyDigest = String;
-type EvaluationResult = serde_json::Value;
 
 #[async_trait]
 pub trait PolicyEngine {
     /// The result is a key-value map.
     /// - `key`: the policy id
-    /// - `value`: It will be a tuple. The first element is the digest of
-    /// the policy (using **Sha384**). The second element is the evaluation
-    /// output of the policy.
+    /// - `value`: the digest of the policy (using **Sha384**).
     async fn evaluate(
         &self,
         reference_data_map: HashMap<String, Vec<String>>,
         input: String,
         policy_ids: Vec<String>,
-    ) -> Result<HashMap<String, (PolicyDigest, EvaluationResult)>>;
+    ) -> Result<HashMap<String, PolicyDigest>>;
 
-    async fn set_policy(&mut self, input: SetPolicyInput) -> Result<()>;
+    async fn set_policy(&mut self, policy_id: String, policy: String) -> Result<()>;
 
-    async fn list_policies(&self) -> Result<Vec<PolicyListEntry>>;
+    /// The result is tuples. The first member is the policy id, and the
+    /// second member is the digest of the policy (using **Sha384**).
+    async fn list_policies(&self) -> Result<Vec<(String, PolicyDigest)>>;
```

Comment:
any reason why we replace the structs with tuples?

Reply:
I once think the struct definition is deep inside the lib code, which would make trouble for client developers. Maybe we can move the definition into only `restful-as.rs`, just above the api handler function. Hopefully which will make the code more readable?

Reply:
> I once think the struct definition is deep inside the lib code, which would make trouble for client developers.

if the fn is public it's possible to [re-export](https://doc.rust-lang.org/reference/items/use-declarations.html#use-visibility) such a struct to make it appear on the module of the fn:

```rust
pub use crate::some::where::deep::PolicyListEntry;
```

Reply:
I wasn't a huge fan of those structs tbh. They seemed a bit unnecessary to me.

Reply:
@mkulke I hope that the current PR looks good to you. we can do some change upon error handling in another PR, as this PR does not focus on/change that behavior. Feel free to give any more suggestions or merge it.

Reply:
👍 I added #394 to track it